mlflow/utils/uri.py

```diff
@@ -105,12 +105,21 @@ def is_fuse_or_uc_volumes_uri(uri):
     )
 
 
-def is_uc_volumes_uri(uri):
+def _is_uc_volumes_path(path: str) -> bool:
+    """
+    Return True if the given path starts with '/Volumes/', '/volumes', '/Volume/', or '/volume/'.
+
+    Reserved paths for volumes: https://docs.databricks.com/en/volumes/index.html#reserved-paths-for-volumes
+    """
+    return re.match(r"^/[vV]olumes?/", path) is not None
```

Comment:
Is the typo-friendly reserved name paths only for user inputs that will then get coerced into the standard `/Volumes/` / `dbfs:/Volumes/` pathing, or will those reserved declarations be used within the UC volumes product? 
I'm wondering if we need complex optional validation for verification of a tmp directory that is used for copying during registration?

Reply:
> we need complex optional validation

you mean `_is_uc_volumes_path(local_model_dir)`?

Reply:
yep! When registering from a given path in UC Volumes, will UC take a mispelled user input like:

`/volume/db/schema/path` and convert it to `/Volumes/db/schema/path` or will it preserve what the user entered in perpetuity? 

If they coerce, then the internal local mapping wouldn't need the fuzzy typo-friendly regex logic. If they preserve the typo in those reserved namespaces, then we definitely need it. 

It doesn't HURT to have the regex in there with the optional fuzzy match - I was just curious if we NEED it. :)

Reply:
I'd keep the regex! We could consider canonicalizing the volume path (e.g. `/volume/` -> `/Volumes/`).